File: 551-student-attendance-record-i/student-attendance-record-i.java
class Solution {
    public boolean checkRecord(String s) {
        HashMap<Character,Integer> mp=new HashMap<>();
        for(char ch:s.toCharArray()){
            mp.put(ch,mp.getOrDefault(ch,0)+1);
        }
        if(mp.containsKey('A') && mp.get('A')>=2){
            return false;
        }
        int count=0;
        for(int i=1;i<s.length()-1;i++){
            if(s.charAt(i-1)=='L' && s.charAt(i)=='L' && s.charAt(i+1)=='L'){
                count++;
            }
        }
        if(count>0){
            return false;
        }
        return true;
    }
}

File: 551-student-attendance-record-i/student-attendance-record-i.cpp
class Solution {
public:
    bool checkRecord(string s) {
        int A=0;
        int L=0;
        for(int i=0;i<s.length();i++){
            if(s[i]=='A'){
                A++;
            }
            if((i>=1 && s[i-1]=='L') && s[i]=='L' && (i<s.length()-1 && s[i+1]=='L')){
                L++;
            }
        }
        if(A<2 && L==0){
            return true;
        }
        return false;
    }
};